core/src/main/java/io/kestra/core/services/ExecutionService.java

```
@@ -957,4 +957,27 @@ public Trigger resetExecution(FlowWithSource flow, Execution execution, Trigger
 
         return trigger.resetExecution(flowWithDefaults, execution, conditionContext);
     }
+
+    public List<ResolvedTaskRun> resolveTaskRuns(Flow flow, List<TaskRun> taskRuns) {
+        
+        return taskRuns.stream()
+            .map(taskRun -> {
+                try {
+                    // Look up the corresponding task definition
+                    Task task = flow.findTaskByTaskId(taskRun.getTaskId());
+
+                    ResolvedTask resolvedTask = ResolvedTask.of(task);
+
+                    return ResolvedTaskRun.builder()
+                        .taskRun(taskRun)
+                        .resolvedTask(resolvedTask)
+                        .build();
+                } catch (InternalException e) {
```

Please let it bubble up and make the method `throws InternalException`  to avoid try/cath inside the stream you can wrap the lambda inside the `map` method with a `throwFunction()` as we often did